### PR TITLE
Extensibility: Possibility to search for resources in the List widgets

### LINKED ISSUE #1666: Extensibility: Details View: Table Widget: Search
**Description**
Make the search on the Table Widget functional. Perhaps search in the first column of the table or even better if it is possible to specify in which column you want to search perhaps like `<column>: <string>`? 

**Reasons**
I would be like to be able to search the contents of the table using the search box on the top right of the table widget. Some of our table views have over 400 elements (CVEs, reports, etc) and it makes it hard to find what you are looking for. Once Table paging is working for extensibility tables, it wouldn't even be possible to find the element you're looking for using the Ctrl+F search of the browser.

### PR DESCRIPTION
**Pull request by Lyczeq**
**Description**

Changes proposed in this pull request:

- add a search feature to the `list`
- add a search feature to the `Table` and `ResourceList` widgets
- add tests
- add docs
- remove `showSearchField` feature
- add search usage to the pizzas examples

**Related issue(s)**
Resolves #1666

### CODE REVIEW

**Review by mrCherry97 — APPROVED** [review]
/woof

**Review by mrCherry97 — DISMISSED** [review]
/woof

File: docs/extensibility/display-section.md [hunk 1]
```diff
@@ -502,6 +505,9 @@ ResourceList widgets render a list of Kubernetes resources. The ResourceList wid
...
   - **default** - optional flag. If set to `true`, the list view is sorted by this value by default.
   - **compareFunction** - optional [JSONata](https://docs.jsonata.org/overview.html) compare function. It is required to use `$first` and `$second` variables when comparing two values. There is a special custom function [compareStrings](jsonata.md#comparestringsfirst-second) used to compare two strings, for example, `$compareStrings($first, $second)`
+- **search** - optional search option. It's an array of objects that allows you to search for resources including the value from the given **source**.
+  - **source** - _[required]_ contains a [JSONata](https://docs.jsonata.org/overview.html) expression used to fetch data for the column. In its simplest form, it's the path to the value.
+  - **searchFunction** - optional [JSONata](https://docs.jsonata.org/overview.html) search function. It allows to use `$input` variable to get the search input's value that can be used to search for more complex data.
```
Comment (grego952) [inline docs/extensibility/display-section.md:510]:
```suggestion
  - **searchFunction** - optional [JSONata](https://docs.jsonata.org/overview.html) search function. It allows you to use the `$input` variable to get the search input's value that can be used to search for more complex data.
```

File: docs/extensibility/display-section.md [hunk 2]
```diff
@@ -572,10 +583,11 @@ Table widgets display array data as rows of a table instead of free-standing com
...
-- **showSearchField** - an optional boolean which disables displaying the search input.
 - **sort** - optional sort option. If set to `true`, it allows you to sort using this value. Defaults to false. It can also be set to an object with the following properties:
   - **default** - optional flag. If set to `true`, the list view is sorted by this value by default.
   - **compareFunction** - optional [JSONata](https://docs.jsonata.org/overview.html) compare function. It is required to use `$first` and `$second` variables when comparing two values. There is a special custom function [compareStrings](jsonata.md#comparestringsfirst-second) used to compare two strings, for example, `$compareStrings($first, $second)`
+- **search** - optional search option. If set to `true`, it allows you to search the resource list by this value. Defaults to false. It can also be set to an object with the following property:
+  - **searchFunction** - optional [JSONata](https://docs.jsonata.org/overview.html) search function. It allows to use `$input` variable to get the search input's value that can be used to search for more complex data.
```
Comment (grego952) [inline docs/extensibility/display-section.md:590]:
```suggestion
  - **searchFunction** - optional [JSONata](https://docs.jsonata.org/overview.html) search function. It allows you to use the `$input` variable to get the search input's value that can be used to search for more complex data.
```

**Review by grego952 — APPROVED** [review]

**Review by mrCherry97 — APPROVED** [review]
/meow

**Pull request merged**